### DISCUSSION TITLE
NullPointerException: Error when adding HelloworldApp to my scenario

### DISCUSSION THREAD

**Discussion by SafrasIqbal** [eclipse/mosaic#88] [Q&A]
Hello Everyone, 
I followed the instruction from Basics [Create a ’Hello world’ application based on Maven ](https://www.eclipse.org/mosaic/docs/develop_applications/#create-a-hello-world-application-based-on-maven). 

1. Created the pom.xml and HelloWorldApp.java file as per the recommended folder structure. 
2. Created the jar file. 
3. Copy the jar file to the application directory of my scenario. 
4. Edited the mapping file and added the application to the vehicle as ""applications": [ "HelloWorldApp-0.0.1.jar.HellloWorldApp" ] " . (Not sure is this the fully qualified name for the application path )
5. Run the scenario and got Null Pointer exception error 
`<pre><font color="#4E9A06"><b>safras-linux@safraslinux</b></font>:<font color="#3465A4"><b>~/mosaic/bundle/target</b></font>$ ./mosaic.sh -s notts -v
2021-04-24 01:27:55,099 INFO  ROOT - Running Eclipse MOSAIC 21.1-SNAPSHOT on Java JRE v11.0.10 (Ubuntu)
2021-04-24 01:27:55,452 INFO  FederationManagement - Start federation with id &apos;notts&apos;
2021-04-24 01:27:55,454 INFO  FederationManagement - Add ambassador/federate with id &apos;application&apos;
2021-04-24 01:27:55,459 INFO  FederationManagement - Add ambassador/federate with id &apos;mapping&apos;
2021-04-24 01:27:55,460 INFO  FederationManagement - Add ambassador/federate with id &apos;sumo&apos;
2021-04-24 01:27:55,461 INFO  FederationManagement - Deploying federate &apos;sumo&apos; locally in ./tmp/sumo
2021-04-24 01:27:55,494 INFO  FederationManagement - Starting federate &apos;sumo&apos; locally in ./tmp/sumo
2021-04-24 01:27:55,495 INFO  FederationManagement - Add ambassador/federate with id &apos;output&apos;
01:28:05 - Simulating: 0ns (0.0s / 1000.0s) - 0.0% (RTF:0.00, ETC:unknown)                        
'-------------------------------------------------------------------
 Stopping simulation due to a critical error:
	- InternalFederateException: Error code: 0x01000013: There was an error during an advance time.
	- Root Cause: NullPointerException: 
 Please see the log files for details.
	- Log-Directory: logs/log-20210424-012754-notts

 'Eclipse MOSAIC will now shutdown.'
-------------------------------------------------------------------
</pre>`
Anyone please help how to fix this issue. I suspect I made a mistake in step 4. 
Thank you in advance.

**Comment by kschrab** [comment]
Have you already looked into MOSAIC.log or the other log files? You can also call MOSAIC with the parameter `-o DEBUG` to get more information about the error.

**Reply by SafrasIqbal** [reply]
Hi @kschrab , thank you for the reply.  see below for the output with -o DEBUG. 
![Screenshot from 2021-04-26 10-57-14](https://user-images.githubusercontent.com/18607009/116064807-45149900-a67e-11eb-9db8-b38568506797.png)
Below is the Mosaic log file. 
[MOSAIC.log](https://github.com/eclipse/mosaic/files/6376103/MOSAIC.log)
Appreciate your help on understanding this issue.  Thanks

**Reply by SafrasIqbal** [reply]
Also other logs if its helps. thanks
[Navigation.log](https://github.com/eclipse/mosaic/files/6376270/Navigation.log)
[Traffic.log](https://github.com/eclipse/mosaic/files/6376273/Traffic.log)
[Mapping.log](https://github.com/eclipse/mosaic/files/6376274/Mapping.log)
[Application.log](https://github.com/eclipse/mosaic/files/6376275/Application.log)
[Communication.log](https://github.com/eclipse/mosaic/files/6376276/Communication.log)
[RuntimeEvents.csv](https://github.com/eclipse/mosaic/files/6376277/RuntimeEvents.csv)

**Reply by kschrab** [reply]
Ah, I just saw in your initial question that you used the wrong name for the application. In Java, the "full-qualified class name" is the name of the class with the package name as suffix, thus you should use `"HellloWorldApp"` only (unless you put your class into any package).

**Reply by SafrasIqbal** [reply]
Cool Great , I tried this before it gave an error as Class not found , it was because typo in the package name ` "HellloWorldApp"` -> `"HelloWorldApp" `:facepalm: . an extra l. you have enlighten me with your comment.  Thank you. It is working now :rocket:
